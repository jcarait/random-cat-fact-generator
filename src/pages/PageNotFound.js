import image404 from "../images/404.jpg";

export default function PageNotFound({ errorMessage }) {
  return (
    <div className="bg-page-not-found">
      <div className="image-container">
        <img
          className="image-404"
          src={image404}
          alt="cat holding sign with label 404"
        ></img>
        {errorMessage && (
          <p className="error-message" data-testid="error">
            Error Message: {errorMessage}{" "}
          </p>
        )}

        <a
          className="attribution"
          href="https://www.freepik.com/vectors/unavailable"
        >
          Unavailable vector created by pikisuperstar - www.freepik.com
        </a>
      </div>
    </div>
  );
}
