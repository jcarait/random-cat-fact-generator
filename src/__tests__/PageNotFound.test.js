import React from "react";
import PageNotFound from "../pages/PageNotFound";
import renderer from "react-test-renderer";

describe("<PageNotFound />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<PageNotFound errorMessage={"error message"} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
