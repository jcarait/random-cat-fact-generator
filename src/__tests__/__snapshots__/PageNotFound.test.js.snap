// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> renders correctly 1`] = `
<div
  className="bg-page-not-found"
>
  <div
    className="image-container"
  >
    <img
      alt="cat holding sign with label 404"
      className="image-404"
      src="404.jpg"
    />
    <p
      className="error-message"
      data-testid="error"
    >
      Error Message: 
      error message
    </p>
    <a
      className="attribution"
      href="https://www.freepik.com/vectors/unavailable"
    >
      Unavailable vector created by pikisuperstar - www.freepik.com
    </a>
  </div>
</div>
`;
